### PR TITLE
add documentation comments

### PR DESCRIPTION
**Pull request by antoinekm**
**What did I change?**

I've added API documentation.

**Why did I change it?**

I noticed that the code was perhaps a little complicated to understand for a beginner who would like to implement the package in his personal project or CRON task, so I added comments and documentation thanks to paka.dev. The comments will appear on paka as soon as the package is released.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +119,28 @@ Here's an example of what you should expect:
 > - You can run the script as many times as you want. It will only index the pages that are not already indexed.
 > - Sites with a large number of pages might take a while to index, be patient.
 
+## API
```
Comment (goenning) [inline README.md]: I think people might get confused and think this is a Web API, maybe we replace it with one of these?
```
## Programmatic Usage
## Using as NPM Module
```

?

Comment (antoinekm) [reply README.md]: Totally agree, I'll sort out the conflicts and change that!

**Comment by antoinekm** [comment]
It should be better now @goenning

**Comment by goenning** [comment]
🚀

**Pull request merged**